### PR TITLE
mbdben ethereum-types upgrade 

### PR DESCRIPTION
**Pull request by rakita**
#249
Upgraded ethereum-types and other libs:

ethereum-types -> 0.9.2
rlp -> 0.4.6
keccak-hash -> 0.5.0
parity-crypto -> 0.6.2
ethabi -> 12.0.0
ethabi-derive -> 12.0.0
ethabi-contract -> 11.0.0
ethbloom -> 0.9.1
rand -> 0.7.3
trie-standardmap -> 0.15.2
triehash -> 0.5.0

### CODE REVIEW

**Review by karim-agha — APPROVED** [review]

File: crates/ethcore/src/miner/stratum.rs [hunk 1]
```diff
@@ -47,6 +47,14 @@ pub struct Options {
     pub secret: Option<H256>,
 }
 
+fn clean_0x(s: &str) -> &str {
```
Comment (jomuel) [inline crates/ethcore/src/miner/stratum.rs:50]: This function is added twice in the PR, here and in bin/oe/helpers, could it be moved to utils?

**Review by jomuel — APPROVED** [review]
Why are we replacing standard `HeapSizeOf` with parity `MallocSizeOf`? `HeapSizeOf` should be derive-able too?

**Comment by rakita** [comment]
> Why are we replacing standard `HeapSizeOf` with parity `MallocSizeOf`? `HeapSizeOf` should be derive-able too?
Servo stopped maintaining HeapSizeOf and we are using their internal lib MallocSizeOf: https://github.com/servo/heapsize

File: crates/ethcore/src/miner/stratum.rs (hunk 1 above)
Comment (rakita) [reply crates/ethcore/src/miner/stratum.rs:50]: It is a one `if` function, and I am fine with it for now.

**Pull request merged**